### PR TITLE
21523 csv download

### PR DESCRIPTION
**Pull request by dan-ai**
adding csv download

### CODE REVIEW

File: post-payments.php [hunk 1]
```diff
@@ -191,7 +198,40 @@ public function format_currency( $number ) {
 		return $this->currency_symbol . number_format( $number, 2 );
 	}
 
-	
+	public function download_report() {
+
+		if ( ! empty( $_GET['download'] ) && 'download_report' == $_GET['download'] ) {
+			// restricting to admins only
+			if ( ! current_user_can( 'update_core' ) ) {
```
Comment (joshkadis) [inline post-payments.php]: maybe make this capability a property of the class? I can see where someone might want an easy way to adjust it

File: post-payments.php [hunk 2]
```diff
@@ -191,7 +198,40 @@ public function format_currency( $number ) {
...
+				wp_die( __( 'You do not have permission to do this', 'post-payments' ) );
+			}
+			// setting headers
+			header('Content-Type: text/csv; charset=utf-8');
+			header('Content-Disposition: attachment; filename=' . sanitize_text_field( $_GET['from_date'] ) . '-to-' . sanitize_text_field( $_GET['to_date'] ) . '-author-data.csv');
+			// getting authors for report
+			$authors = $this->get_report_data( sanitize_text_field( $_GET['from_date'] ), sanitize_text_field( $_GET['to_date'] ) );
+			// start creating the csv string
+			$csv = "Total,Name,Articles\n";
```
Comment (joshkadis) [inline post-payments.php]: translation

File: post-payments.php [hunk 3]
```diff
@@ -191,7 +198,40 @@ public function format_currency( $number ) {
...
+			foreach ( $authors as $name => $author ) {
+				$author_posts = $author['posts'];
+				foreach ( $author_posts as $key => $post ) {
+					if ( 0 == $key ) {
+						// first row for each author
+						$csv .= $author['total'] . ',' . $name . ',' . get_the_title( $post ) . "\n";
```
Comment (joshkadis) [inline post-payments.php]: escaping

File: post-payments.php [hunk 4]
```diff
@@ -191,7 +198,40 @@ public function format_currency( $number ) {
...
+					} else {
+						// empty info to group articles by author
+						$csv .= '' . ',' . '' . ',' . get_the_title( $post ) . "\n";
```
Comment (joshkadis) [inline post-payments.php]: escaping

**Comment by joshkadis** [comment]
some minor stuff then :pizza: yourself

File: post-payments.php (hunk 1 above)
Comment (xmattus) [inline post-payments.php]: Yeah, agreed.

File: post-payments.php [hunk 5]
```diff
@@ -191,7 +198,40 @@ public function format_currency( $number ) {
...
-	
+	public function download_report() {
+
+		if ( ! empty( $_GET['download'] ) && 'download_report' == $_GET['download'] ) {
+			// restricting to admins only
+			if ( ! current_user_can( 'update_core' ) ) {
+				wp_die( __( 'You do not have permission to do this', 'post-payments' ) );
+			}
+			// setting headers
+			header('Content-Type: text/csv; charset=utf-8');
```
Comment (xmattus) [inline post-payments.php]: spacing around parens here

File: post-payments.php [hunk 6]
```diff
@@ -191,7 +198,40 @@ public function format_currency( $number ) {
...
+			header('Content-Disposition: attachment; filename=' . sanitize_text_field( $_GET['from_date'] ) . '-to-' . sanitize_text_field( $_GET['to_date'] ) . '-author-data.csv');
```
Comment (xmattus) [inline post-payments.php]: and here

**Comment by xmattus** [comment]
Also minor code style issues. Otherwise nice work!

**Comment by joshkadis** [comment]
:pig:

**Pull request merged**